stacker/config/__init__.py

```diff
@@ -35,5 +80,202 @@ def parse_config(raw_config, environment=None):
+
+    branch = StringType(serialize_when_none=False)
+
+    commit = StringType(serialize_when_none=False)
+
+    paths = ListType(StringType, serialize_when_none=False)
+
+
+class PackageSources(Model):
+    git = ListType(ModelType(GitPackageSource))
+
+
+class Hook(Model):
+    path = StringType(required=True)
+
+    required = BooleanType(default=True)
+
+    data_key = StringType(serialize_when_none=False)
+
+    args = DictType(AnyType)
+
+
+class Stack(Model):
+    name = StringType(required=True)
+
+    class_path = StringType(required=True)
+
+    requires = ListType(StringType, serialize_when_none=False)
+
+    locked = BooleanType(default=False)
+
+    enabled = BooleanType(default=True)
+
+    variables = DictType(AnyType, serialize_when_none=False)
+
+    parameters = DictType(AnyType, serialize_when_none=False)
+
+    def validate_parameters(self, data, value):
+        if value:
+            stack_name = data['name']
+            raise ValidationError(
+                    "Stack definition %s contains deprecated "
+                    "'parameters', rather than 'variables'. Please "
+                    "update your config." % stack_name)
+        return value
+
+
+class Config(Model):
+    """This is the Python representation of a stacker config file.
+
+    This is used internally by stacker to parse and validate a yaml formatted
+    stacker configuration file, but can also be used in scripts to generate a
+    stacker config file before handing it off to stacker to build/destroy.
+
+    Example::
+
+        from stacker.config import dump, Config, Stack
+
+        vpc = Stack({
+            "name": "vpc",
+            "class_path": "blueprints.VPC"})
+
+        config = Config()
+        config.namespace = "prod"
+        config.stacks = [vpc]
+
+        print dump(config)
+
+    """
+
+    # TODO(ejholmes): This field should be required, but isn't for backwards
+    # compatibility reasons.
+    namespace = StringType()
+
+    namespace_delimiter = StringType(serialize_when_none=False)
+
+    stacker_bucket = StringType(serialize_when_none=False)
+
+    stacker_bucket_region = StringType(serialize_when_none=False)
+
+    stacker_cache_dir = StringType(serialize_when_none=False)
+
+    sys_path = StringType(serialize_when_none=False)
+
+    package_sources = ModelType(PackageSources, serialize_when_none=False)
+
+    pre_build = ListType(ModelType(Hook), serialize_when_none=False)
+
+    post_build = ListType(ModelType(Hook), serialize_when_none=False)
+
+    pre_destroy = ListType(ModelType(Hook), serialize_when_none=False)
+
+    post_destroy = ListType(ModelType(Hook), serialize_when_none=False)
+
+    tags = DictType(StringType, serialize_when_none=False)
+
+    mappings = DictType(
+        DictType(DictType(StringType)), serialize_when_none=False)
+
+    lookups = DictType(StringType, serialize_when_none=False)
+
+    stacks = ListType(ModelType(Stack), default=[], validators=[more_than_one])
```

Comment:
Would probably be good to add a validator on here that ensures no stacks have the same name (e.g. you accidentally overwrite an existing stack, I've done this).